File: robocode-tankroyale-bot-api-dotnet-core/events/BotHitWallEvent.cs
namespace Robocode.TankRoyale
{
  /// <summary>
  /// Event occurring when the bot has hit a wall.
  /// </summary>
  public class BotHitWallEvent : Event
  {
    /// <summary>ID of the victim bot that hit the wall.<summary>
    int VictimId { get; }

    /// <summary>
    /// Constructor.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="victimId">ID of the victim bot that hit the wall.</param>
    public BotHitWallEvent(int turnNumber, int victimId) : base(turnNumber) => VictimId = victimId;
  }
}

File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/BotHitWallEvent.java
package dev.robocode.tankroyale.botapi.events;

/** Event occurring when your bot has hit a wall. */
@SuppressWarnings("unused")
public final class BotHitWallEvent extends Event {

  /**
   * Initializes a new instance of the BotHitWallEvent class.
   *
   * @param turnNumber is the turn number when the bot has hit the wall.
   */
  public BotHitWallEvent(int turnNumber) {
    super(turnNumber);
  }
}
